### PR TITLE
Updated logging.py for #9146

### LINKED ISSUE #9146: Documentation for caplog `set_level()` should make it clearer it's a threshold
#### What's the problem this feature will solve?

The [documentation for `caplog.set_level()`](https://docs.pytest.org/en/stable/reference.html#pytest.LogCaptureFixture.set_level) says
> Set the level of a logger for the duration of a test.

This does not make it clear that it's setting the _threshold_ level for capturing logging.

#### Describe the solution you'd like

The Python logging documentation makes this clear for `logging.setLevel()`:
> Sets the threshold for this logger to level. Logging messages which are less severe than level will be ignored; ...

`caplog.set_level()` ought to do the same.

#### Additional context

The only reason I know it's a threshold is because I tried running `pytest -rA` with different levels set in caplog and the output contained messages at that level and above

### PR DESCRIPTION
**Pull request by Vijay-Arora**
Close #9146

### CODE REVIEW

**Comment by nicoddemus** [comment]
Tip: you can add `Closes #9146` to the PR description, which serves both to link the PR to the issue, and GitHub will close the issue automatically after the PR is merged (just added it myself). 👍

File: changelog/9146.doc.rst
Comment (nicoddemus) [inline changelog/9146.doc.rst:1]: No need to add the issue number to the description, this will be added automatically later. Better to describe the change itself.

```suggestion
Improve documentation for `caplog.set_level`.
```

File: src/_pytest/logging.py
Comment (nicoddemus) [inline src/_pytest/logging.py:453]:
```suggestion
        """
        Sets the threshold for this logger to level. 
        
        Logging messages which are less severe than level will be ignored; logging messages 
        which have this severity level or higher will be emitted by whichever handler or handlers service
        this logger, unless a handler’s level has been set to a higher severity level than level.
```

**Comment by Vijay-Arora** [comment]
Thanks! But I see there are some checks which I see failed for eg. At least 1 approving review is required by reviewers with write access. What should I do in this case? Should I update the PR or let it be?

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Thanks @Vijay-Arora! Left a few request for changes, please take a look.

cc @MetRonnie

File: changelog/9146.doc.rst
Comment (Vijay-Arora) [reply changelog/9146.doc.rst:1]: Thank you!

**Comment by Vijay-Arora** [comment]
> Thanks @Vijay-Arora! Left a few request for changes, please take a look.
> 
> cc @MetRonnie

Hey @nicoddemus, should I go ahead and make the requested changes and open a new PR for this?

**Comment by nicoddemus** [comment]
> Hey @nicoddemus, should I go ahead and make the requested changes and open a new PR for this? 

Just push the commits to the same branch and the PR will update automatically.

> Also, I see some failures, do I need to make some changes for that too?

Hmm ignore them for now, they don't seem related to your PR, I will investigate those later.

**Comment by Vijay-Arora** [comment]
> > Hey @nicoddemus, should I go ahead and make the requested changes and open a new PR for this?
> 
> Just push new commits to the same branch and the PR will update automatically.
> 
> > Also, I see some failures, do I need to make some changes for that too?
> 
> Hmm ignore them for now, they don't seem related to your PR, I will investigate those later.

Cool, thanks, Making the changes now.

File: src/_pytest/logging.py
Comment (MetRonnie) [inline src/_pytest/logging.py:450]: In my opinion, the original tagline was slightly clearer, but just needed to mention it's a threshold.
```suggestion
        """Set the threshold level of a logger for the duration of a test.
```

File: src/_pytest/logging.py
Comment (MetRonnie) [inline src/_pytest/logging.py:455]: I'm not sure some of these details from the logging module make sense in the context of caplog. My suggestion is keep it short and simple 😃 
```suggestion
        Logging messages which are less severe than this level will not be captured.
```

**Review by MetRonnie — COMMENTED** [review]
Thanks for taking this on! As the author of the original issue, I've added my thoughts

File: src/_pytest/logging.py
Comment (nicoddemus) [reply src/_pytest/logging.py:455]: Agree, good points. @Vijay-Arora could you implement the change plase?

**Comment by MetRonnie** [comment]
@nicoddemus I do wonder if the docstring for [`at_level`](https://docs.pytest.org/en/stable/reference.html#pytest.LogCaptureFixture.at_level) should be updated at the same time, but I don't know whether the behaviour is also a threshold or not, I've never used it.

File: src/_pytest/logging.py
Comment (Vijay-Arora) [reply src/_pytest/logging.py:455]: Hi @nicoddemus and @MetRonnie , Thanks for the feedback; I'll definitely make the changes.

**Comment by ziebam** [comment]
@MetRonnie Good catch. As far as I understand it, the behavior is the same for [`at_level`](https://docs.pytest.org/en/stable/reference/reference.html#pytest.LogCaptureFixture.at_level) as it also utilizes the [`setLevel`](https://docs.python.org/3/library/logging.html#logging.Handler.setLevel) method, therefore an update for its docs seems reasonable.

@Vijay-Arora Would you be interested in driving this one home? 😄

File: src/_pytest/logging.py
Comment (thisch) [reply src/_pytest/logging.py:450]: @Vijay-Arora could you implement the change please?

**Review by Zac-HD — APPROVED** [review]
Let's get this merged!

**Pull request merged**
